fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -783,13 +784,39 @@ private void computeFragmentExecParams() throws Exception {
                         params.instanceExecParams.size() + destParams.perExchNumSenders.get(exchId.asInt()));
             }
 
-            // add destination host to this fragment's destination
-            for (int j = 0; j < destParams.instanceExecParams.size(); ++j) {
-                TPlanFragmentDestination dest = new TPlanFragmentDestination();
-                dest.fragment_instance_id = destParams.instanceExecParams.get(j).instanceId;
-                dest.server = toRpcHost(destParams.instanceExecParams.get(j).host);
-                dest.setBrpcServer(toBrpcHost(destParams.instanceExecParams.get(j).host));
-                params.destinations.add(dest);
+            if (bucketShuffleJoinController.isBucketShuffleJoin(destFragment.getFragmentId().asInt())) {
```

Comment:
If you could abstract the query schedule strategy like `Presto` and refactor this class, which would be very great.

Reply:
Yes, It is the future work to do. But this part is complex and has a great impact. I think we need to this work gradually

Reply:
OK